packages/core/src/domain/character-sanitizer.ts

```diff
@@ -0,0 +1,42 @@
+import { Address } from "./address";
+import { Patient } from "./patient";
+
+/**
+ * Regex to match invalid UTF-8 characters that should be stripped from patient data.
+ * Currently targets the replacement character (�) which indicates corrupted encoding.
+ */
+const badCharactersRegex = /[�]/g;
+
+/**
+ * Strips invalid UTF-8 characters from a string.
+ *
+ * @param str - The string to sanitize
+ * @returns The sanitized string with invalid characters removed, or undefined if input is falsy
+ */
+function stripBadCharactersFromString(str: string): string {
+  return str.replace(badCharactersRegex, "");
+}
+
+/**
+ * Strips invalid UTF-8 characters from patient address data.
+ *
+ * @param patient - The patient whose data should be sanitized
+ * @returns A new patient object with sanitized address data
+ */
+export function stripInvalidCharactersFromPatientData(patient: Patient): Patient {
```

Comment:
nit: name this to be specific to address? Or update the TSDoc?

BTW, don't we want to also run it for first/last names?

Reply:
Agree, maybe `stripInvalidCharactersFromPatientAddress` unless it should be extended to sanitizing the name string as well? It might be worth sanitizing the name, since [some states also allow diacritics in names](https://en.wikipedia.org/wiki/Naming_in_the_United_States#Names_with_diacritics_or_non-English_letters).

Reply:
addressed - yeah just ran it over first + last name too.